File: 백준/Silver/10799. 쇠막대기/쇠막대기.js
'use strict';
const path = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const input = require('fs').readFileSync(path).toString().trim().split('\n');

const brackets = input.join('');
let result = 0;

const stack = [];
for (let i = 0; i < brackets.length; i++) {
  if (brackets[i] === '(') {
    stack.push(brackets[i]);
  } else {
    stack.pop();
    if (brackets[i - 1] === '(') {
      // 레이저인 경우
      result += stack.length;
    } else {
      // ))인 경우
      result++;
    }
  }
}

console.log(result);

File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
import sys

# sys.stdin = open("input.txt", "r")

input = sys.stdin.readline 

data = input()
stack = []
result = 0

for  i  in range(len(data)):
  if data[i] == '(':
    stack.append(data[i])    
  elif data[i] == ')':
    stack.pop()
    if data[i - 1] == '(':
      result += len(stack)
    else:
      result += 1

print(result)

